### PR TITLE
Make TestKeeper_HandleRelay more scalable

### PR DESCRIPTION
**Pull request by msmania**
This will help #1536 to add new testcases with the tolerance.

### CODE REVIEW

**Comment by Olshansk** [comment]
@msmania Should we also rename `ClientSessionBlockSyncAllowance` to `ClientSessionSyncAllowance` in this PR so we can merge the other one?

**Comment by nodiesBlade** [comment]
I can cherry pick these changes into my branch as well - whatever works best. @Olshansk .

Also tests look nicer here for sure, nice work @msmania !

File: x/pocketcore/keeper/service_test.go [hunk 1]
```diff
@@ -12,79 +15,180 @@ import (
...
+	ctx, keeper, kvkeys, clientPrivateKey, appPrivateKey, nodePubKey, chain :=
+		setupHandleRelayTest(t)
 
-	mockCtx.On("KVStore", keeper.storeKey).Return(ctx.KVStore(keeper.storeKey))
-	mockCtx.On("KVStore", keys["pos"]).Return(ctx.KVStore(keys["pos"]))
-	mockCtx.On("KVStore", keys["params"]).Return(ctx.KVStore(keys["params"]))
-	mockCtx.On("KVStore", keys["application"]).Return(ctx.KVStore(keys["application"]))
+	SessionRangesToTest := 2
+	// Eliminate the impact of ClientBlockSyncAllowance
+	types.GlobalPocketConfig.ClientBlockSyncAllowance = 10000
```
Comment (Olshansk) [inline x/pocketcore/keeper/service_test.go:124]: Revert in t.cleanup?

File: x/pocketcore/keeper/service_test.go [hunk 2]
```diff
@@ -12,79 +15,180 @@ import (
...
+	)
 	assert.Nil(t, err, err)
 	assert.NotNil(t, resp)
 	assert.NotEmpty(t, resp)
 	assert.Equal(t, resp.Response, "bar")
+
+	// Client is behind or advanced beyond Node's height
+	//   --> CodeInvalidBlockHeightError
+	for i := 1; i <= SessionRangesToTest; i++ {
+		resp, err = testRelayAt(
```
Comment (Olshansk) [inline x/pocketcore/keeper/service_test.go:165]:
```suggestion
		// Fail on relay that's too far in the past
		resp, err = testRelayAt(
```

File: x/pocketcore/keeper/service_test.go [hunk 3]
```diff
@@ -12,79 +15,180 @@ import (
...
+			clientPrivateKey,
+			appPrivateKey,
+			nodePubKey,
+			chain,
+		)
+		assert.Nil(t, resp)
+		assert.NotNil(t, err)
+		assert.Equal(t, err.Codespace(), sdk.CodespaceType(types.ModuleName))
+		assert.Equal(t, err.Code(), sdk.CodeType(types.CodeInvalidBlockHeightError))
+		resp, err = testRelayAt(
```
Comment (Olshansk) [inline x/pocketcore/keeper/service_test.go:179]:
```suggestion
		// Fail on relay that's in the future
		resp, err = testRelayAt(
```

**Review by Olshansk — APPROVED** [review]
A few minor nits but feel free to merge after adressing.

**Comment by Olshansk** [comment]
> I can cherry pick these changes into my branch as well - whatever works best. @Olshansk .
> 
> Also tests look nicer here for sure, nice work @msmania !

@PoktBlade Let's merge this one in and you can rebase/merge with it in your branch.

**Comment by msmania** [comment]
> I can cherry pick these changes into my branch as well - whatever works best. @Olshansk .

That would be great.  I'm fine with closing this.  Thanks!

**Pull request merged**